### ISSUE TITLE
Python3 incompatibility

### ISSUE THREAD

**Issue by boykebunk** [EBI-Metagenomics/mobilome-annotation-pipeline#36]
While running 
process > RENAME (1)  
an error comes up regarding the python env:
Command error:
  /usr/bin/env: ‘python’: No such file or directory

Most presumably the pipeline is dependent on the servers python environment although not mentioned in the dependencies list. In our case we have installed python3 on the servers, which is not recognized by /usr/bin/env python, but by /usr/bin/env python3, which now should be standard on every new system.

I would recommend to run python2 scripts (if absolutely needed) within a singularity container or to convert them to python3.

**Comment by mberacochea** [comment]
Hi @boykebunk,

All the tools, or at least the majority of them, have a corresponding container. The RENAME process uses a [container with Biopython](https://github.com/EBI-Metagenomics/mobilome-annotation-pipeline/blob/main/modules/rename_contigs.nf#L8).

Additionally, all the scripts use Python 3, as Python 2 has long been deprecated. Most modern systems have Python 3 as the default Python installation.

Cheers

**Comment by boykebunk** [comment]
Hi @mberacochea 
ok, good to hear. Then I am still wondering why the error occurs. May I attach the nextflow.log?
[.nextflow.log](https://github.com/user-attachments/files/16758974/default.nextflow.log)

**Comment by mberacochea** [comment]
You should use the [singularity](https://docs.sylabs.io/guides/3.0/user-guide/installation.html) profile. I'll amend the documentation (but it's mentioned in the [setup instructions](https://github.com/EBI-Metagenomics/mobilome-annotation-pipeline?tab=readme-ov-file#install-and-dependencies)), as it's lacking that in the nextflow run examples.

In your case it should be:
```bash
$ nextflow run /home/bob10/mobilome-annotation-pipeline/main.nf \
--assembly HEG6_1234.contigs_plasmids.fasta \
-profile singularity
```

ICEfinder container needs to be built by the users, as per the LICENSE we can't distribute it with the pipeline.

Let me know if you bump into more issues with the pipeline.

**Comment by boykebunk** [comment]
Hi, I tried to build the ICEfinder container today, but ran into the following error:
`/.post.script: 18: cpanm: not found
FATAL:   While performing build: while running engine: exit status 127`

**Comment by boykebunk** [comment]
Hi @mberacochea 
after having successfully build the ICE finder container I tried to run again the pipeline. I used:
`nextflow run /home/bob10/mobilome-annotation-pipeline/main.nf --assembly SWE2_1234.contigs_plasmids.fasta --icefinder_sif /home/bob10/mobilome-annotation-pipeline/templates/icefinder-v1.0-local.sif
`
-profile singularity does not work for me, probably I did not find the correct section within the instructions to build. I am attaching the [.nextflow.log](https://github.com/user-attachments/files/17027746/default.nextflow.log) again.
